packages/@aws-cdk/aws-ec2/lib/network-acl.ts

```diff
@@ -0,0 +1,411 @@
+   *
+   * @default 17
+   */
+  readonly protocol: number;
+
+  /**
+   * Whether to allow or deny traffic that matches the rule; valid values are "allow" or "deny".
+   */
+  readonly ruleAction: string
+
+  /**
+   * Rule number to assign to the entry, such as 100. ACL entries are processed in ascending order by rule number.
+   * Entries can't use the same rule number unless one is an egress rule and the other is an ingress rule.
+   */
+  readonly ruleNumber: number
+}
+
+export class NetworkAclEntry extends NetworkAclEntryBase {
+  public readonly  cidrBlock: string;
+  public readonly  egress: boolean;
+  public readonly  icmp?: Icmp;
+  public readonly  ipv6CidrBlock?: string;
+  public readonly  networkAcl: INetworkACL;
+  public readonly  portRange?: PortRange;
+  public readonly  protocol: number;
+  public readonly  ruleAction: string;
+  public readonly  ruleNumber: number;
+
+  constructor(scope: Construct, id: string, props: NetworkAclEntryProps) {
+    super(scope, id, {
+      physicalName: props.networkAclEntryName
+    });
+
+    this.cidrBlock = props.cidrBlock;
+    this.egress = props.egress;
+    this.icmp = props.icmp;
+    this.networkAcl = props.networkAcl;
+    this.ipv6CidrBlock = props.ipv6CidrBlock;
+    this.portRange = props.portRange;
+    this.protocol = props.protocol;
+    this.ruleAction = props.ruleAction;
+    this.ruleNumber = props.ruleNumber;
+
+    new CfnNetworkAclEntry(this, 'Resource', {
+      networkAclId: this.networkAcl.networkAclId,
+      ruleNumber: props.ruleNumber,
+      protocol: props.protocol,
+      ruleAction: props.ruleAction,
+      egress: props.egress,
+      cidrBlock: props.cidrBlock,
+      icmp: props.icmp,
+      portRange: props.portRange
+    });
+
+  }
+
+  public get uniqueId() {
+    return this.node.uniqueId;
+  }
+}
+
+export interface ISubnetNetworkAclAssociation extends IResource {
+  /**
+   * ID for the current SubnetNetworkAclAssociation
+   * @attribute
+   */
+  readonly subnetNetworkAclAssociationAssociationId: string;
+}
+
+/**
+ * A SubnetNetworkAclAssociationBase that is not created in this template
+ */
+abstract class SubnetNetworkAclAssociationBase extends Resource implements ISubnetNetworkAclAssociation {
+public abstract readonly subnetNetworkAclAssociationAssociationId: string;
+constructor(scope: Construct, id: string, props?: ResourceProps) {
+    super(scope, id, props);
+
+  }
+
+  public get uniqueId() {
+    return this.node.uniqueId;
+  }
+}
+
+export interface SubnetNetworkAclAssociationProps {
+  /**
+   * The name of the SubnetNetworkAclAssociation.
+   *
+   * It is not recommended to use an explicit name.
+   *
+   * @default If you don't specify a SubnetNetworkAclAssociationName, AWS CloudFormation generates a
+   * unique physical ID and uses that ID for the group name.
+   */
+  readonly subnetNetworkAclAssociationName?: string;
+
+  /**
+   * ID for the current Network ACL
+   * @attribute
+   */
+  readonly networkAclId: string;
```

Comment:
Should I define this as INetworkACL instead of string , as this Construct is now defined